clarity/src/vm/analysis/type_checker/contexts.rs

```diff
@@ -86,6 +86,25 @@ impl TypeMap {
         }
     }
 
+    /// Like set_type but forcing a change if already set
+    pub fn override_type(&mut self, expr: &SymbolicExpression, type_sig: TypeSignature) {
```

Comment:
Is "overriding" the correct term here? I have the impression that the correct term would be "overwrite" since we completely replace the previous type.